Sources/ClientRuntime/Networking/Streaming/BufferedStream.swift

```diff
@@ -107,4 +131,21 @@ public class BufferedStream: Stream {
             isClosed = true
         }
     }
+
+    // If there is a client waiting to read data, and there is unread data remaining in the buffer,
+    // read the data and pass it to the reader via the continuation.
+    // Continue until there are no clients or no data.
+    private func _flushToNextClient() {
+        while !clients.isEmpty {
```

Comment:
We should wrap this in the lock, to prevent any bugs from human error where someone may call this function without using a lock.